crates/matrix-sdk-base/src/response_processors/room/sync_v2.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2025 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use ruma::{api::client::sync::sync_events::v3::JoinedRoom, OwnedRoomId, OwnedUserId, RoomId};
+use tokio::sync::broadcast::Sender;
+
+#[cfg(feature = "e2e-encryption")]
+use super::super::e2ee;
+use super::super::{account_data, ephemeral_events, state_events, timeline, Context};
+use crate::{
+    store::{ambiguity_map::AmbiguityCache, BaseStateStore},
+    sync::JoinedRoomUpdate,
+    RequestedRequiredStates, Result, RoomInfoNotableUpdate, RoomState,
+};
+
+/// Process updates of a joined room.
+#[allow(clippy::too_many_arguments)]
+pub async fn update_joined_room(
```

Comment:
Probably missing something here but can't we name all the processor main methods `process_fo` e.g. `process_joined_room` in this case? 🤔 
`update_foo` makes me think it updates something in place where it actually returns an update.

Reply:
Ah, got it, fair enough!